kuwala/core/canvas/src/components/Modals/TransformationCatalog/TransformationCatalogModal.js

```diff
@@ -0,0 +1,344 @@
+                color={'kuwalaRed'}
+                icon={catalogItem.icon}
+                text={catalogItem.name}
+            />
+        )
+    }
+
+    const CatalogBody = () => {
+        return (
+            <div className={Classes.CatalogBodyContainer}>
+                {
+                    selectedTransformationIndex !== null
+                    ?
+                        <SelectedTransformationCatalog/>
+                    :
+                        <div className={Classes.CatalogBodyAlertText}>
+                            <p>To pick a transformation block,</p>
+                            <p>select a transformation <span className={'text-kuwala-red'}>category</span> first</p>
+                        </div>
+                }
+            </div>
+        )
+    }
+
+    const SelectedTransformationCatalog = () => {
+        if (selectedTransformationIndex === null || catalogOptions.length === 0) {
+            return (
+                <div>
+                    No <span className={'text-kuwala-red'}>transformation</span> options found.
+                </div>
+            )
+        } else {
+            return (
+                <div className={Classes.SelectedTransformationContainer}>
+                    <div className={Classes.TransformationOptionsContainer}>
+                        {renderTransformationOptions(catalogOptions)}
+                    </div>
+                    <div className={Classes.SelectedOptionDetailsContainer}>
+                        <div className={Classes.OptionDetailsContainer}>
+                            {renderSelectedOptionDetails()}
+                        </div>
+                    </div>
+                </div>
+            )
+        }
+    }
+
+    const renderTransformationOptions = (options) => {
+        return options.map(renderOptionItem)
+    }
+
+    const renderOptionItem = (optionItem, index) => {
+        return (
+            <Button
+                solid={false}
+                color={'kuwalaPurple'}
+                onClick={()=>{
+                    setSelectedCatalogOption(index)
+                }}
+                selected={selectedCatalogOption === index}
+                icon={optionItem.icon}
+                text={optionItem.name}
+            />
+        )
+    }
+
+    const renderSelectedOptionDetails = () => {
+        if(selectedCatalogOption === null){
+            return (
+                <div className={Classes.OptionDetailsAlertText}>
+                    <p>To see the <span className={'text-kuwala-purple'}>details</span> of a transformation block,</p>
+                    <p>select one from the left</p>
+                </div>
+            )
+        } else {
+            const optionItem = catalogOptions[selectedCatalogOption];
+            return (
+                <div className={Classes.OptionDetailsContent}>
+                    <div className={'text-kuwala-purple space-x-4'}>
+                        {getCatalogItemIcon(optionItem.icon)}
```

Comment:
Something seems to be wrong here with the icons.

In the detail view of a transformation catalog item, there is still the old "default" icon you've used for some transformations (`compare_with_date`, `add_columns`, `compare_with_number`, `is_true`, `is_false`).

<img width="296" alt="image" src="https://user-images.githubusercontent.com/28758040/166217778-7c355234-4d21-4a13-b585-96453da9113b.png">